### PR TITLE
DOC: fix Dataframe.to_records examples

### PR DESCRIPTION
**Pull request by H0R5E**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Fixed the examples of Dataframe.to_records related to timestamp conversion following the changes in default behaviour made in pull request #18902. As requested in issue #18160.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1380,25 +1380,33 @@ def to_records(self, index=True, convert_datetime64=None):
...
 
-        The timestamp conversion can be disabled so NumPy's datetime64
-        data type is used instead:
+        Setting convert_datetime64 to False provides the default behaviour:
 
         >>> df.to_records(convert_datetime64=False)
         rec.array([('2018-01-01T09:00:00.000000000', 1, 0.5 ),
                    ('2018-01-01T09:01:00.000000000', 2, 0.75)],
                   dtype=[('index', '<M8[ns]'), ('A', '<i8'), ('B', '<f8')])
+
```
Comment (datapythonista) [inline pandas/core/frame.py]: As `convert_datetime64` is now deprecated, I'd get rid of the examples for it. I feel like they encourage writing code that will be broken soon.

Also, do you mind removing the blank lines before and after the closing `"""` of the docstring?

**Review by datapythonista — COMMENTED** [review]
Thanks for the fixes. If you want, you can also replace in this PR, `boolean` by `bool` in the parameters types, remove the `y` in `Returns` and just leave the type, and add a description of what is returned in the next line.

This way the docstring will follow all the standards.

Thanks!

**Comment by H0R5E** [comment]
Changes made, as requested.

I also added an example of what happens when the index of the dataframe is given a label. As expected, the attribute of the recarray is named using the label of the index, but I thought it was ambiguous from the docs what might happen.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -1374,31 +1377,20 @@ def to_records(self, index=True, convert_datetime64=None):
         rec.array([('a', 1, 0.5 ), ('b', 2, 0.75)],
                   dtype=[('index', 'O'), ('A', '<i8'), ('B', '<f8')])
 
+        If the DataFrame index has no label then the recarray field name
+        is set to 'index'. If the index has a label then this is used as the
+        field name:
+
+        >>> df.index.rename("I", inplace=True)
```
Comment (jreback) [inline pandas/core/frame.py]: don't use inplace in examples, pls just reassign the index

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 2 above)
Comment (H0R5E) [reply pandas/core/frame.py]: Done!

File: ci/doctests.sh [hunk 3]
```diff
@@ -21,7 +21,7 @@ if [ "$DOCTEST" ]; then
 
     # DataFrame / Series docstrings
     pytest --doctest-modules -v pandas/core/frame.py \
-        -k"-assign -axes -combine -isin -itertuples -join -nlargest -nsmallest -nunique -pivot_table -quantile -query -reindex -reindex_axis -replace -round -set_index -stack -to_dict -to_records -to_stata -transform"
+        -k"-assign -axes -combine -isin -itertuples -join -nlargest -nsmallest -nunique -pivot_table -quantile -query -reindex -reindex_axis -replace -round -set_index -stack -to_dict -to_stata -transform"
```
Comment (jreback) [inline ci/doctests.sh]: is this on-purpose?

Comment (H0R5E) [reply ci/doctests.sh]: I was asked to make this change by @TomAugspurger in issue #18160. I believe the point is to check that the examples are correct.

Comment (TomAugspurger) [reply ci/doctests.sh]: This is a whitelist of doctests that are allowed to fail till we fix them.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @H0R5E 

@pandas-dev/pandas-core merge on green.

**Comment by H0R5E** [comment]
Happy to help. Keep up the great work!

File: ci/doctests.sh (hunk 3 above)
Comment (jorisvandenbossche) [reply ci/doctests.sh]: Yep, going to open an issue about this

**Review by datapythonista — APPROVED** [review]

**Comment by datapythonista** [comment]
Thanks for the contribution @H0R5E

**Pull request merged**